### PR TITLE
Install sxs with the install-dotnet scripts

### PR DESCRIPTION
**Pull request by ZEisinger**
Fixes: #25 

Using the install-dotnet scripts to inside in default global location to enable SxS builds and to use cached versions on hosted agents by default. Install-dotnet.ps1 script has been updated to support proxy setting requirements.

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ jobs:
   build:
     runs-on: ${{ matrix.operating-system }}
     strategy:
+      fail-fast: false
```
Comment (StanleyGoldman) [inline .github/workflows/workflow.yml:14]: Curious question: What does `fail-fast` do?

**Review by StanleyGoldman — APPROVED** [review]

File: .github/workflows/workflow.yml (hunk 1 above)
Comment (ZEisinger) [reply .github/workflows/workflow.yml:14]: When fail-fast is true the other jobs in the matrix are immediately cancelled when one of the job fails. So when the MacOS job fails, the Windows job (which typically takes longer) is immediately cancelled and potentially hides different errors.

**Comment by jetersen** [comment]
Docs should be updated to include an sxs example.

**Comment by ZEisinger** [comment]
@jetersen Thanks for calling this out, I've added an example to README: https://github.com/actions/setup-dotnet#usage

**Pull request merged**